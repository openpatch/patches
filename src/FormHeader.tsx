import { css } from "@emotion/react";
import { Heading } from "./Heading";
import { Box } from "./Box";
import { Divider } from "./Divider";
import { ReactNode } from "react";

export type FormHeaderProps = {
  id?: string;
  children?: ReactNode;
};

export const FormHeader = ({ id, children }: FormHeaderProps) => {
  return (
    <Box
      mb="large"
      id={id}
      css={css`
        :target:before {
          content: "";
          display: block;
          height: 100px;
          margin: -100px 0 0;
        }
      `}
    >
      <Heading
        mb="xsmall"
        as="h3"
        css={css`
          display: flex;
          align-items: center;
        `}
      >
        {children}
      </Heading>
      <Divider color="neutral.100" />
    </Box>
  );
};
