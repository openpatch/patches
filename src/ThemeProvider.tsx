import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
