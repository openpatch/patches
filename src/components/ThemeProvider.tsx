import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
