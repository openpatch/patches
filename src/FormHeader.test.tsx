
import { render } from "@testing-library/react";
import { FormHeader, FormHeaderProps } from "./FormHeader";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: FormHeaderProps) =>
render(
    <ThemeProvider>
        <FormHeader {...props} />
    </ThemeProvider>
);

test("should render", () => {
   renderWithTheme();
});

test("should match snapshot", () => {
    const { container } = renderWithTheme();
    expect(container).toMatchSnapshot();
});