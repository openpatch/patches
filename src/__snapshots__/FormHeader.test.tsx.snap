// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  margin-bottom: 32px;
}

.emotion-0:target:before {
  content: "";
  display: block;
  height: 100px;
  margin: -100px 0 0;
}

.emotion-1 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  padding: 0px;
  margin: 0px;
  margin-bottom: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  position: relative;
}

.emotion-3 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  position: absolute;
  width: 100%;
  background-color: #E4E7EB;
  height: 1px;
}

<div>
  <div
    id="patches-root"
  >
    <div
      class="emotion-0"
    >
      <h3
        class="emotion-1"
      />
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
          height="1px"
          width="100%"
        />
      </div>
    </div>
  </div>
</div>
`;
